File: CSharpFundamentals/CSharpAdvanced/StacksAndQueues_Ex/BasicStackOperations.cs
using System;
using System.Collections.Generic;

namespace BasicStackOperations
{
    class BasicStackOperations
    {
        private static Stack<int> stack = new Stack<int>();
        static void Main(string[] args)
        {
            string[] tokens = Console.ReadLine().Split();
            int N = int.Parse(tokens[0]);
            int S = int.Parse(tokens[1]);
            int X = int.Parse(tokens[2]);
            string[] nums = Console.ReadLine().Split();
            for (int i = 0; i < N; i++)
            {
                stack.Push(int.Parse(nums[i]));
            }
            for (int i = 0; i < S; i++)
            {
                stack.Pop();
            }
            if (stack.Contains(X))
            {
                Console.WriteLine("true");
            }
            else
            {
                int min = int.MaxValue;
                if (stack.Count == 0)
                {
                    min = 0;
                }
                while (stack.Count > 0)
                {
                    int local = stack.Pop();
                    if (local < min)
                    {
                        min = local;
                    }
                }
                Console.WriteLine(min);
            }
        }
    }
}

File: JavaFundamentals/JavaAdvanced/Stacks-and-queues-ex/src/BasicStackOperations.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayDeque;
import java.util.Arrays;

public class BasicStackOperations {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        int[] input = Arrays.stream(reader.readLine().split("\\s+"))
                .mapToInt(Integer::parseInt).toArray();
        int elementsToPush = input[0];
        int elementsToPop = input[1];
        int elementToCheck = input[2];

        int[] elements = Arrays.stream(reader.readLine().split("\\s+"))
                .mapToInt(Integer::parseInt).toArray();

        if (elementsToPush == elementsToPop) {
            System.out.println(0);
            return;
        }

        ArrayDeque<Integer> stack = new ArrayDeque<>();

        for (int i = 0; i < elementsToPush; i++) {
            stack.push(elements[i]);
        }

        for (int i = 0; i < elementsToPop; i++) {
            stack.pop();
        }

        int min = Integer.MAX_VALUE;

        while (!stack.isEmpty()){
            int number = stack.pop();
            if (number == elementToCheck){
                System.out.println("true");
                return;
            }
            if (number < min){
                min = number;
            }
        }

        System.out.println(min);
    }
}
